rolling-update-kubernetes/src/test/scala/akka/rollingupdate/kubernetes/AppVersionRevisionSpec.scala

```diff
@@ -0,0 +1,187 @@
+      namespace = Some(namespace),
+      namespacePath = "",
+      podName = podName,
+      secure = false,
+      apiServiceRequestTimeout = 2.seconds,
+      customResourceSettings = new CustomResourceSettings(enabled = false, crName = None, 60.seconds)
+    )
+  }
+
+  private val kubernetesApi =
+    new KubernetesApiImpl(
+      system,
+      settings(podName1),
+      namespace,
+      apiToken = "apiToken",
+      clientHttpsConnectionContext = None)
+
+  override implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(timeout = Span(5, Seconds), interval = Span(100, Millis))
+
+  override protected def afterAll(): Unit = super.shutdown()
+
+  override protected def beforeEach(): Unit = {
+    wireMockServer.resetAll()
+  }
+
+  private def podPath(podName: String) =
+    urlEqualTo(s"/api/v1/namespaces/$namespace/pods/$podName")
+
+  private def replicaPath(replica: String) =
+    urlEqualTo(s"/apis/apps/v1/namespaces/$namespace/replicasets/$replica")
+
+  private def getPod(podName: String): MappingBuilder =
+    get(podPath(podName)).withHeader("Content-Type", new EqualToPattern("application/json"))
+
+  private def getReplicaSet(replica: String): MappingBuilder =
+    get(replicaPath(replica)).withHeader("Content-Type", new EqualToPattern("application/json"))
+
+  private val defaultPodResponseJson =
+    """{
+      | "metadata": {
+      |   "ownerReferences": [
+      |     {"name": "wrong-replicaset-id", "kind": "SomethingElse"},
+      |     {"name": "parent-replicaset-id", "kind": "ReplicaSet"}
+      |    ]
+      |  }
+      |}""".stripMargin
+
+  private val defaultReplicaResponseJson =
+    """{
+      | "metadata": {
+      |   "annotations": {
+      |     "deployment.kubernetes.io/revision": "1"
+      |   }
+      |  }
+      |}""".stripMargin
+
+  private def stubPodResponse(json: String = defaultPodResponseJson) =
+    stubFor(
+      getPod(podName1).willReturn(
+        ResponseDefinitionBuilder.okForJson("").withJsonBody(mapper.readTree(json))
+      ))
+
+  private def stubReplicaResponse(json: String = defaultReplicaResponseJson) =
+    stubFor(
+      getReplicaSet("parent-replicaset-id").willReturn(
+        ResponseDefinitionBuilder.okForJson("").withJsonBody(mapper.readTree(json))
+      ))
+
+  "Read revision from Kubernetes" should {
+
+    "parse pod and replica response to get the revision" in {
+      stubPodResponse()
+      stubReplicaResponse()
+
+      EventFilter
+        .info(pattern = "Reading revision from Kubernetes: akka.cluster.app-version was set to 1", occurrences = 1)
+        .intercept {
+          kubernetesApi.readRevision()
+        }
+    }
+
+    "retry and then fail when pod not found" in {
+      stubFor(getPod(podName1).willReturn(aResponse().withStatus(404)))
+      EventFilter
+        .warning(pattern = ".*Failed to get revision", occurrences = 5)
+        .intercept({
+          kubernetesApi.readRevision()
+        })
```

Comment:
Verify that the returned future is failed also?

Reply:
Jea! Or event if the right version is returned 😄